physics/mp_thompson.F90

```diff
@@ -495,19 +408,26 @@ subroutine mp_thompson_run(ncol, nlev, con_g, con_rd,        &
             return
          end if
 
-         !> - Convert specific humidity/moist mixing ratios to dry mixing ratios
-         qv_mp = spechum/(1.0_kind_phys-spechum)
-         qc_mp = qc/(1.0_kind_phys-spechum)
-         qr_mp = qr/(1.0_kind_phys-spechum)
-         qi_mp = qi/(1.0_kind_phys-spechum)
-         qs_mp = qs/(1.0_kind_phys-spechum)
-         qg_mp = qg/(1.0_kind_phys-spechum)
-
-         !> - Convert number concentrations from moist to dry
-         ni_mp = ni/(1.0_kind_phys-spechum)
-         nr_mp = nr/(1.0_kind_phys-spechum)
-         if (is_aerosol_aware) then
-            nc_mp = nc/(1.0_kind_phys-spechum)
+         !> - Convert specific humidity to water vapor mixing ratio.
+         !> - Also, hydrometeor variables are mass or number mixing ratio
+         !> - either kg of species per kg of dry air, or per kg of (dry + vapor).
+
+         qv = spechum/(1.0_kind_phys-spechum)
+
+         if (convert_dry_rho) then
+           qc = qc/(1.0_kind_phys-spechum)
+           qr = qr/(1.0_kind_phys-spechum)
+           qi = qi/(1.0_kind_phys-spechum)
+           qs = qs/(1.0_kind_phys-spechum)
+           qg = qg/(1.0_kind_phys-spechum)
+
+           ni = ni/(1.0_kind_phys-spechum)
+           nr = nr/(1.0_kind_phys-spechum)
+           if (is_aerosol_aware) then
+              nc = nc/(1.0_kind_phys-spechum)
+              nwfa = nwfa/(1.0_kind_phys-spechum)
+              nifa = nifa/(1.0_kind_phys-spechum)
+           end if
          end if
```

Comment:
@climbfuji Is convert_dry_rho a new namelist parameter? Is it .true. for the GSD physics suite?

Reply:
Currently `convert_dry_rho` is defined as a module variable in `physics/mp_thompson.F90` at the top. The value is `.False.`

Reply:
Do we need to consider covert_dry_rho in GFS_suite_interstitial and GFS_rrtmg_pre?

Reply:
This variable is host-model specific, so I think yes. I should update this PR and move the `convert_dry_rho` to `GFS_typedefs.F90` and give it a CCPP standard name. Let me do that later today ...

Reply:
I definitely like the idea of elevating a boolean variable for whether or not some subroutines should deal with the conversion between moist and dry air units of microphysics species.  It will generalize the framework better for adoption of different models.